### ISSUE TITLE
bug: Error "JSX syntax extension is not currently enabled"

### ISSUE THREAD

**Issue by mryechkin** [favware/esbuild-plugin-version-injector#64]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the bug

I'm getting an error when using this plugin in a project that has `loader: { '.js': 'jsx' }` option set in esbuild:

<img width="1077" alt="Screenshot 2023-07-11 at 2 12 27 PM" src="https://github.com/favware/esbuild-plugin-version-injector/assets/5657181/34db5f9f-4e3b-49a0-b3ad-a99e49236aed">

Here's my full esbuild config for reference:

```jsx
import esbuild from 'esbuild';
import { esbuildPluginVersionInjector } from 'esbuild-plugin-version-injector';

const args = process.argv.slice(2);
const watch = args.includes('--watch');

const options = {
  banner: {
    js: `'use client'`,
  },
  bundle: true,
  color: true,
  entryPoints: ['./src/index.js'],
  external: ['./node_modules/*', 'react', 'react-dom', 'styled-components'],
  format: 'esm',
  loader: {
    '.js': 'jsx',
  },
  logLevel: 'info',
  outfile: '.esbuild/lib.js',
  plugins: [esbuildPluginVersionInjector()],
  sourcemap: process.env.WATCH_MODE === 'true',
};

const ctx = await esbuild.context(options);

try {
  if (watch) {
    // Run watch mode
    await ctx.watch();
  } else {
    // Build and exit
    await esbuild.build(options);
    await ctx.dispose();
  }
} catch (error) {
  console.error(error);
  process.exit(1);
}
```

### Steps To Reproduce

1. Create a JavaScript project with esbuild configured to bundle a single ESM file
2. Set the `loader` option to `'.js': 'jsx'`
3. Add this plugin and try to run the build

### Expected behavior

Build should run with no errors

### Screenshots

<img width="1077" alt="Screenshot 2023-07-11 at 2 12 27 PM" src="https://github.com/favware/esbuild-plugin-version-injector/assets/5657181/171a1c7d-e1be-47ca-8952-743b7a5c8556">

**Comment by favna** [comment]
Do you have JSX code in files ending in `.jsx`? If so, you should change that to `.jsx` file extension. This library has code in place to ensure that the proper loader is used, but it's based on file extensions:
https://github.com/favware/esbuild-plugin-version-injector/blob/65127c8f842fe5c1b0670622b6053e4a4b55c8d8/src/index.ts#L109-L132

I can do an analysis if I get anything else from `OnLoadArgs` with options like you configured, but please provide a small GitHub repository with sample code then.

**Comment by mryechkin** [comment]
> Do you have JSX code in files ending in .js instead of .jsx? If so, you should change that to .jsx file extension.

Yes, that's why I have that `loader` option set. I'm well aware of the recommended conventions, but it's not an option for me in this library at this time.

If I understand correctly, is there no way to use this plugin in this case then? I would've expected the `loader` option to apply for the plugin as well.

**Comment by favna** [comment]
Released v1.2.0 with the option `disableOnLoadTrigger`. If you set that to `true` then I believe you can use this plugin again.
